presto-main/src/test/java/com/facebook/presto/operator/scalar/TestRegexpFunctions.java

```diff
@@ -141,6 +141,22 @@ public void testRegexpReplace()
         assertFunction("REGEXP_REPLACE(VARCHAR 'x', '.*', 'xxxxx')", createUnboundedVarcharType(), "xxxxxxxxxx");
     }
 
+    @Test
+    public void testRegexpReplaceLambda()
+    {
+        assertFunction("REGEXP_REPLACE('new', '(\\w)', x -> x[1])", createUnboundedVarcharType(), "new");
```

Comment:
add test that produces `newNEW`, e.g, lambda:
```
x -> x[1] || upper(x[2])
```

Reply:
Did you mean for pattern '(\\w)'? This pattern matches single char so it'll be producing "nNeEwW" :
REGEXP_REPLACE('new', '(\\w)', x -> x[1] || upper(x[2])) = "nNeEwW"

However I added this case:
REGEXP_REPLACE('new', '(\\w*)', x -> x[1] || upper(x[2]))" = "newNEW"